test/test_input.py

```diff
@@ -172,34 +257,98 @@ def test_input_keywords(self):
         self.assertTrue(np.isclose(qkw.evaluate()[0], np.eye(3)).all())
 
         # Failure case (wrong argument type)
-        with self.assertRaises(RuntimeError):
+        with self.assertRaises(RuntimeError) as err:
             InputKeywords["zeeman"]([], args=["wrong"])
+        self.assertEqual(
+            str(err.exception),
+            "Error parsing keyword argument(s) 'zeeman': "
+            "invalid literal for int() with "
+            "base 10: 'wrong'",
+        )
 
-    def test_input(self):
+    def test_read_block(self):
```

Comment:
This is a general comment, but particularly obvious in this long block here: unit tests should ideally be as small as possible and test only one thing (hence the "unit"). This block tests multiple import files – each file could be broken up into its own test with a descriptive name. e.g.
```
test_input_valid()
test_input_invalid_formatting()
test_input_invalid_keyword()
...
```
You can get away with pretty long names since these functions aren't used anywhere else. Notice how with a good descriptive name and single-unit tests, you can get an idea of the problem just from seeing the name of the test that's failed, without having to look through the test logs. Breaking things up also stops any failures from being hidden by an earlier failure within the same function.

I'd especially recommend breaking up these blocks which test file inputs, but the lark tests could be broken up too.